main/src/ca/uwaterloo/flix/language/ast/KindedAst.scala

```diff
@@ -105,6 +107,8 @@ object KindedAst {
 
     case class LetRec(sym: Symbol.VarSym, ann: Annotations, mod: Ast.Modifiers, exp1: Expr, exp2: Expr, loc: SourceLocation) extends Expr
 
+    case class LocalDef(ann: Annotations, sym: Symbol.VarSym, fparams: List[FormalParam], declaredTvar: Type, declaredEvar: Type, exp1: Expr, exp2: Expr, loc: SourceLocation) extends Expr
```

Comment:
Ascribe calls them `ExpectedType` and `ExpectedPur`. Not great names but I think the two should be consistent